### PR TITLE
Specify tox ver. to not use 4.0.0+ to correct some python package tests

### PR DESCRIPTION
**Pull request by sameluch**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
The following python packages are using tox as a part of their package tests. The 3.27.1 version (or higher) is being specified as a breaking change was introduced in tox v4.0.0 to the way some variables are handled.
pyproject-rpm-macros
python-coverage
python-nocasedict
python-pbr
python-prettytable
python-pynacl
python-tqdm
python-twisted
python-virtualenv

###### Change Log  

- Specified tox version for pip install as >=3.27.1 and < 4.0.0 to allow for patches to the 3.X.X version to be taken

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- x86 buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=278979&view=results
- ARM64 buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=278980&view=results

### CODE REVIEW

File: SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec [hunk 1]
```diff
@@ -7,7 +7,7 @@ Name:           pyproject-rpm-macros
 #   Increment Z when this is a bugfix or a cosmetic change
 # Dropping support for EOL Fedoras is *not* considered a breaking change
 Version:        1.0.0~rc1
-Release:        3%{?dist}
+Release:        4%{?dist}
```
Comment (mfrw) [inline SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec:10]: The toolchain-manfiests need a bump for `pyproject-rpm-macros`

**Review by mfrw — APPROVED** [review]
LGTM, but please bump the toolchain-manifests for `pyproject`

File: SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec [hunk 2]
```diff
@@ -117,6 +117,9 @@ test_status=$?
 %{_rpmconfigdir}/mariner/pyproject_requirements_txt.py
 
 %changelog
+* Fri Dec 16 2022 Sam Meluch <sammeluch@microsoft.com> - 1.0.0-rc1-4
```
Comment (PawelWMS) [inline SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec:120]:
```suggestion
* Fri Dec 16 2022 Sam Meluch <sammeluch@microsoft.com> - 1.0.0~rc1-4
```

**Review by PawelWMS — COMMENTED** [review]
Looks good, just needs that manifest update mentioned by Falak and one typo fix.

File: SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec (hunk 1 above)
Comment (sameluch) [reply SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec:10]: Manifest files updated.

File: SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec (hunk 2 above)
Comment (sameluch) [reply SPECS/pyproject-rpm-macros/pyproject-rpm-macros.spec]: Version number corrected

**Review by PawelWMS — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**